### PR TITLE
Add helper function trace.Start (fixes #1090)

### LINKED ISSUE #1090: Suggested helper function `trace.Start(ctx context.Context, name string, opts ...StartOption) (context.Context, Span)`
### Suggestion:

The suggestion is to add a helper function `trace.Start` to `api/trace` to aid the happy path of creating a child span within a function. The function would look something like:
```go
func Start(ctx context.Context, name string, opts ...StartOption) (context.Context, Span)
  parentSpan := trace.SpanFromContext(ctx)
  tracer := parentSpan.Tracer()
  return tracer.Start(ctx, name, opts...)
}
```

### Reasoning:

In converting some go apps that use opencensus to opentelemetry, one thing that I've found to be a sticking point in the transition is the happy path where a function takes a context and wants to add a span underneath, e.g. in opencensus you can do:

```go
import (
  "go.opencensus.io/trace"
)

func foo(ctx context.Context) error {
  ctx, span := trace.StartSpan(ctx, "child")
  defer span.End()

  // ...
}
```
whilst in opentelemetry you either have to repeat `tracer := global.TraceProvider().Tracer("example.com/trace")` in every package/function with a shared state of the tracer name, or since the context contains the span, and the span links to the tracer I found myself doing this a lot:

```go
import (
  "go.opentelemetry.io/otel/api/trace"
)

func foo(ctx context.Context) error {
  parentSpan := trace.SpanFromContext(ctx)
  tracer := parentSpan.Tracer()
  ctx, span := tracer.Start(ctx, "child")
  defer span.End()

  // ...
}
```
In coming to opentelemetry I found this "one step beyond hello-world example" a stumbling block and a tad repetitive. I would suggest a helper function like the above would aid readability, and ease first-time user experience.

It also means that modules using this pattern can be imported by two distinct applications and it will extract their individually named tracers without having to share the tracer name across the two applications and library (e.g. `tracer := global.TraceProvider().Tracer("shared-trace-name")`). If the importing or calling code doesn't use opentelemetry, and the context contains no span then the span extraction [falls back to NoopSpan](https://github.com/open-telemetry/opentelemetry-go/blob/master/api/trace/context.go#L39) and causes no runtime issues, so functions can be coded to be "opentelemetry-ready" even if the caller hasn't hooked in an exporter yet.

I would love to know others' opinions on if this would be a useful usability feature, and I'm happy to code up a PR if people think this is a good idea.

### PR DESCRIPTION
**Pull request by pci**
This is an implementation of https://github.com/open-telemetry/opentelemetry-go/issues/1090 with the motivation for the change in that issue. I've tried to follow the test patterns of the surrounding tests as best I could. The test case internally uses `ContextWithSpan` and `SpanFromContext` so it's not a true unit test of solely that function, but it seemed the best way to test the functionality.

### CODE REVIEW

**Comment by pci** [comment]
I've signed a CLA, and not quite sure why it's saying it can't find my UserID on that commit.

**Comment by pci** [comment]
There we go, was an issue with multiple email addresses. Sorted now.

**Comment by Aneurysm9** [comment]
The one thing that worries me about this approach is that it takes away knowledge of the `TracerProvider` and `Tracer` used to create the new `Span`.  I could see this potentially being a problem if middleware had established its own `TracerProvider` with a different set of `SpanProcessor`s and exporting to a different location than the global `TracerProvider` that the user is likely to expect.  In that case if the `Span` in the current `Context` was created from that middleware `TracerProvider` then the newly created `Span` may not export to the location the user expects or may be processed in unexpected ways.

Would it be preferable to use the global `TracerProvider` to get a `Tracer` rather than extracting the `Tracer` from the `Span` contained in the context?

**Comment by pci** [comment]
@Aneurysm9 Ah, I think I've misunderstood a key difference between OpenCensus and OpenTelemetry, and I opened the underlying 'issue' 7 months ago and best practice has matured since then.

In OpenCensus each tracer had to be wired up to an exporter (in practical usage there was only ever one tracer), so in coming to OpenTelemetry my assumption was that to get cohesive trace outputs in your exporter that either all packages would have to use the same tracer, or each package would have to export its tracer so that the main entry code could wire them all up to the exporter. And if you used multiple tracer that the outputs wouldn't be linked in the final output of the TraceProvider.

Whereas reading some of the more recent changes it now seems much clearer that I probably misunderstood and that in OpenTelemetry that the main program defines the trace provider and that each package is free to make it's own tracer from that provider and you don't need to worry about passing through or wiring in, and you'll still get linked outputs. For example if you had a main and a sub packages:

foo/main.go
```go
func main() {
    // setup, say, stackdriver exporter `tp`
    otel.SetTracerProvider(tp)
    
    // use default traceprovider
    tr := otel.Tracer("github.com/pci/.../foo")
    ctx, span := tr.Start(ctx, "b")
    defer span.End()
    b.Run(ctx)
}
```

and a second package foo/b/b.go
```go
var (
  // use default traceprovider
  tr = otel.Tracer("github.com/pci/.../foo/b")
)

func Run(ctx context.Context) {
  ctx, otherSpan := tr.Start(ctx, "other_things")
  defer otherSpan.End()
  // do other stuff
}
```

then in, again say, stackdriver you'll still get a cohesive waterfall?

TL;DR - I think I misunderstood opentelemetry best practise.

@Aneurysm9 if you wouldn't mind confirming the above, just in case I got that wrong, but then I'll close this PR and the linked issue. And thank you for your time, much appreciated!

**Comment by MrAlias** [comment]
@pci your comment about best practices looks accurate to me. Does that mean this PR and related Issue can be closed?

**Comment by pci** [comment]
@MrAlias apologies for the laster reply, had a family emergency come up. Yes, now I understand best practise better I think this PR would be at best not-needed, at worse an anti-pattern! I'll close the PR and issue and leave a comment with an explanation. Thanks again.

**Comment by MrAlias** [comment]
> @MrAlias apologies for the late reply, had a family emergency come up. Yes, now I understand best practise better I think this PR would be at best not-needed, at worse an anti-pattern! I'll close the PR and issue and leave a comment with an explanation. Thanks again.

No worries! Thanks for the contribution and feedback :smile:

**Comment by rizalgowandy** [comment]
I am so used to using `opentracing` I also thought the child creation can be automated using just context
without predefined global package variable `tracer`.
Oh well, I guess it should do.
